### DISCUSSION TITLE
HTTP Request state machine

### DISCUSSION THREAD

**Discussion by pke** [statelyai/xstate#1623] [Ideas]
Anyone tried to use xstate to implement https://github.com/for-GET/http-decision-diagram/blob/master/doc/README.md yet?

It seems like a major effort but since the base work is already done its a matter of putting that into xstate terms.

I think this would warrant a community effort and to be release as an xstate-http package with agnostic fetch library that could be handed in (axios, fetch, XMLHttpRequest).

They have a [FSM json](https://github.com/for-GET/http-decision-diagram/blob/master/httpdd.fsm.json) already maybe this could be the base for an xstate machine?

And here in all its glory the HTTP Decision Diagram:

![http-decision-diagram (2)](https://user-images.githubusercontent.com/164625/98738769-7f2c0000-23a8-11eb-8deb-be3c46c7ce5d.png)

**Comment by Andarist** [comment]
That FSM JSON looks really promising - it should be fairly trivial to convert this to the XState schema.
